scanner_cli.py

```diff
@@ -104,6 +104,87 @@ def fetch_properties(state_code):
     except requests.exceptions.RequestException as ex:
         print(f"Error: {ex}")
 
+def parse_mortgage_file(input_file, output_file):
+    # parse mortgage file
+    last_row = []
+
+    # parse zillow file
+    with open(input_file, 'r', newline='', encoding='utf-8') as csvfile:
+        reader = csv.DictReader(csvfile)
+
+        # iterate through each row
+        for row in reader:
+            date = row['DATE']
+            mortgage_rate = row['MORTGAGE30US']
+
+            mortgage_data = {
+                'date': date,
+                'mortgage_rate': float(mortgage_rate)
+            }
+
+            # Update last_row with the current row
+            last_row = mortgage_data
+
+    # Save the parsed data in JSON format
+    with open(output_file, 'w', encoding='utf-8') as jsonfile:
+        json.dump(last_row, jsonfile, indent=2)
+
+    print(f"Parsed data saved to {output_file}")
+
+def parse_zillow_file(input_file, output_file, data_name):
+    data = []
+
+    # parse zillow file
+    with open(input_file, 'r', newline='', encoding='utf-8') as csvfile:
+        reader = csv.DictReader(csvfile)
+
+        region_data = {}
+
+        for row in reader:
+            # grab the last key in the dict
+            last_key = list(row)[-1]
+            if row['RegionName'] != "United States":
+                # seperate region name into state and city
+                city = row['RegionName'].split(", ")[0]
+                state_code = row['RegionName'].split(", ")[1]
+                
+                if state_code in region_data:
+                    region_data[state_code].update({
+                        city: {data_name: row[last_key]}# Grab the most recent data column
+                    })
+                else:
+                    region_data[state_code] = {
+                        row['RegionName'].split(", ")[0]: {data_name: row[last_key]}
+                    }
+
+            data.append(region_data)
+
+    # Save the parsed data in JSON format
+    with open(output_file, 'w', encoding='utf-8') as jsonfile:
+        json.dump(data, jsonfile, indent=2)
+
+    print(f"Parsed data saved to {output_file}")
+
+def parse_external_data():
```

Comment:
when do we make the API calls?

Reply:
there are no API calls with these functions, they are for parsing the data I am getting from CSVs as I couldn't find any API for the other data points